plugin/pkg/admission/priority/admission.go

```diff
@@ -164,6 +164,16 @@ func (p *PriorityPlugin) admitPod(a admission.Attributes) error {
 		return admission.NewForbidden(a, fmt.Errorf("the integer value of priority must not be provided in pod spec. Priority admission controller populates the value from the given PriorityClass name"))
 	}
 	if utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
+		if operation == admission.Update {
+			oldPod, ok := a.GetOldObject().(*api.Pod)
+			if !ok {
+				return errors.NewBadRequest("resource was marked with kind Pod but was unable to be converted")
+			}
+			if oldPod.Spec.PriorityClassName == pod.Spec.PriorityClassName {
+				// if user didn't modify priority in purpose, we don't caculate the new one.
+				return nil
```

Comment:
This doesn't look right to me.  What is stopping me from leaving priorityclassname alone and changing priority directly?

Reply:
> This doesn't look right to me. What is stopping me from leaving priorityclassname alone and changing priority directly?

Oh, and if that is a hole, add a test that checks for it and ensures you can't bypass it like that.

Reply:
Our pod update validation logic allows only several fields in PodSpec to be updated. Priority and PriorityClassName cannot be updated:
https://github.com/bsalamat/kubernetes/blob/master/pkg/api/validation/validation.go#L2910

Reply:
> Our pod update validation logic allows only several fields in PodSpec to be updated. Priority and PriorityClassName cannot be updated:
> https://github.com/bsalamat/kubernetes/blob/master/pkg/api/validation/validation.go#L2910

Wait.  If you can't update the value here, why are you checking for any updates at all?

Reply:
> Wait. If you can't update the value here, why are you checking for any updates at all?

What about:
When user tries to update PriorityClassName or Priority, we return admission.NewForbidden directly?

Reply:
@deads2k As @CaoShuFeng mentioned, the problem here is that if someone updated a field of a Pod, this admission controller would try to resolve the Priority of the Pod again. Now if the PriorityClass was changed and had a different value, the admission controller would update the pod spec with the new value causing an error being thrown by the validation logic. This PR fixes this scenario by not changing the integer value of the pod during updates.

Reply:
> @deads2k As @CaoShuFeng mentioned, the problem here is that if someone updated a field of a Pod, this admission controller would try to resolve the Priority of the Pod again. Now if the PriorityClass was changed and had a different value, the admission controller would update the pod spec with the new value causing an error being thrown by the validation logic. This PR fixes this scenario by not changing the integer value of the pod during updates.

But I think that's all this plugin does.  Why not just stop running any logic here for updates at all?  Around line 117 or so.

Reply:
> But I think that's all this part of the plugin does. Why not just stop running any logic here for updates at all? Around line 117 or so.

I am OK with this. the validation logic will return status code 422 anyway.
What do you think? @bsalamat

Reply:
> But I think that's all this part of the plugin does. Why not just stop running any logic here for updates at all? Around line 117 or so.

I agree. The historical reason for having update monitoring was that we first thought we would allow updating "PriorityClassName" of pods, but we decided not to support it. We can remove the update monitoring.

Reply:
Done.